### ISSUE TITLE
Thread-local lazy static

### ISSUE THREAD

**Issue by jethrogb** [rust-lang-nursery/lazy-static.rs#82]
`lazy_static` and std's `thread_local` are very similar concepts, but `lazy_static`s interface (with `Deref`) is much nicer. I think it should be possible to implement a thread-local version of lazy_static.

**Comment by Kimundi** [comment]
Indeed, some consistency between the two would be nice. I've always thought of that more as changing lazy_statics API to be closer to thread_local, but providing a wrapper for it with a nicer API seems like a interesting idea as well.

**Comment by tuxzz** [comment]
I've made a crate that combined `lazy_static`, `thread_local` and `RefCell`.
https://docs.rs/ref_thread_local/0.0.0/ref_thread_local/
